app/policies/pageflow/entry_policy.rb

```diff
@@ -0,0 +1,241 @@
+                            user_id: user.id])
+      end
+
+      def memberships_above_previewer_for_account_of_entries(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships as pageflow_memberships_2 ON ' \
+                            'pageflow_memberships_2.user_id = :user_id AND ' \
+                            'pageflow_memberships_2.role IN '\
+                            '("editor", "publisher", "manager") AND ' \
+                            'pageflow_memberships_2.entity_id = pageflow_entries.account_id ' \
+                            'AND pageflow_memberships_2.entity_type = "Pageflow::Account"',
+                            user_id: user.id])
+      end
+
+      def memberships_above_editor_for_entries(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships ON ' \
+                            'pageflow_memberships.user_id = :user_id AND ' \
+                            'pageflow_memberships.role IN ("publisher", "manager") AND ' \
+                            'pageflow_memberships.entity_id = pageflow_entries.id AND ' \
+                            'pageflow_memberships.entity_type = "Pageflow::Entry"',
+                            user_id: user.id])
+      end
+
+      def memberships_above_editor_for_account_of_entries(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships as pageflow_memberships_2 ON ' \
+                            'pageflow_memberships_2.user_id = :user_id AND ' \
+                            'pageflow_memberships_2.role IN ("publisher", "manager") AND ' \
+                            'pageflow_memberships_2.entity_id = pageflow_entries.account_id ' \
+                            'AND pageflow_memberships_2.entity_type = "Pageflow::Account"',
+                            user_id: user.id])
+      end
+
+      def memberships_above_publisher_for_entries(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships ON ' \
+                            'pageflow_memberships.user_id = :user_id AND ' \
+                            'pageflow_memberships.role IN ("manager") AND ' \
+                            'pageflow_memberships.entity_id = pageflow_entries.id AND ' \
+                            'pageflow_memberships.entity_type = "Pageflow::Entry"',
+                            user_id: user.id])
+      end
+
+      def memberships_above_publisher_for_account_of_entries(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships as pageflow_memberships_2 ON ' \
+                            'pageflow_memberships_2.user_id = :user_id AND ' \
+                            'pageflow_memberships_2.role IN ("manager") AND ' \
+                            'pageflow_memberships_2.entity_id = pageflow_entries.account_id ' \
+                            'AND pageflow_memberships_2.entity_type = "Pageflow::Account"',
+                            user_id: user.id])
+      end
+
```

Comment:
The above methods differ only in details. Can we reduce duplication here by extracting a?